core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -141,6 +136,18 @@ private[spark] abstract class Task[T](
   def executorDeserializeTime: Long = _executorDeserializeTime
 
   /**
+   * Collect the latest values of accumulators used in this task. If the task failed,
+   * filter out the accumulators whose values should not be included on failures.
+   */
+  def collectAccumulatorUpdates(taskFailed: Boolean = false): Seq[AccumulableInfo] = {
+    if (context != null) {
```

Comment:
none currently. I just didn't want to throw a random NPE if this is somehow called before `run`. I think either doing an explicit assert or this check makes sense, because before calling `run` we have no accumulator updates and so it makes sense to return an empty `Seq`.